Inventory.md

```diff
@@ -0,0 +1,173 @@
+
+| Parameter | Description | Exact equivalent in Haskell Byron Node |
+| ----- |:-------------| --- |
+| genesisBlockHash | Hash of the very first block | ✔ |
+| genesisBlockDate | Start time of the chain. | ✔ |
+| feePolicy | Policy regarding transaction fee. | ❌ |
+| slotLength | Length, in seconds, of a slot. | ✔ |
+| epochLength | Number of slots in a single epoch. | ✔ |
+| txMaxSize | Maximum size of a transaction (soft or hard limit). | ✔ |
+| epochStability | Length of the suffix of the chain considered unstable | ✔ |
+| activeSlotCoefficient | In Genesis/Praos, corresponds to the % of active slots (i.e. slots for which someone can be elected as leader). | ❌ |
+
+Genesis and slotting information should trivially be the same. 
+
+The fee policies work differently. 
+
+Jörmungandr charges `constant + coeff * (inputs + outputs) + optional_cert_specific_fee`
+
+The Haskell Byron node charges `constant + coeff * tx_size_in_bytes`
+
+some links:
+
+- [Wallet BlockchainParameters](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1084-L1102)
+- [cardano-ledger ProtocolParameters](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Update/ProtocolParameters.hs#L32-L62)
+- [wallet use of Jörmungandr fee policy](https://github.com/input-output-hk/cardano-wallet/blob/468e0fb14ed3b50ab3d16ba127f24242a41ed126/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs#L105)
+- [doc of Jörmungandr fee policy: format.md](https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/doc/format.md)
+- [cardano-ledger TxFeePolicy](https://github.com/input-output-hk/cardano-ledger/blob/0ad59ce46b3da8af6f6af97064e7c77cc7c2ac8b/cardano-ledger/src/Cardano/Chain/Common/TxFeePolicy.hs#L41-L59)
+
+
+### Protocol Parameter Updates
+
+Jörmungandr does not fully support protocol parameter updates. If protocol parameters can change, the wallet needs to be aware.
+
+- Jörmungandr: static (updates not implemented)
+- Haskell Byron: static (updates possible, but not likely)
+- Haskell Shelley: dynamic (updates possible, wallet should handle this)
+
+### Blocks
+
+The Jörmungandr wallet uses the `BlockHeader = (slotId, blockHeight, hash, prevHash)` record to identify points on the chain.
+
+The Byron Haskell node only *needs* `Point ByronBlock = Origin | At (slotNo, hash)` to start giving us blocks. The wallet can retrieve `blockHeight` and `prevHash` from block headers. But those two fields are not needed to initiate the chain-sync protocol. Small ❌. 
+
+We might need to convert the genesis point `Origin`, to a `BlockHeader` if we are to persist it in the DB ❌
+
+Byron `SlotNo` counts slots from genesis in a single number. The current Jörmungandr-compatible wallet `SlotId` counts the epoch AND the slot in that epoch. ❌
```

Comment:
This approach (SlotNo) should actually make the handling of protocol updates easier :+1:

Reply:
It is a difference nonetheless. (I tried to separate the differences themselves from my "review" of them. I could have mentioned that in the beginning of the document.)

I don't see why this in particular would make protocol updates easier to handle (for the wallet).

Reply:
Because, to reliably keep track of `SlotId` as we do right now, we would need to maintain a record of protocol parameters to know about the epoch lengths and when they change. The slotting arithmetic becomes tricky because, the "next slot" needs to lookup its corresponding epoch length and makes computation from there. 

If you consider `SlotNo` only, computing the next or previous slot is straightforward and can be done regardless of the epoch length value.

Reply:
> The slotting arithmetic becomes tricky because

I don't think we are actually using any slot arithmetic in the wallet. Only when testing. (E.g search for `slotSucc`).

My view was that as long as we have the current `ProtocolParameters`, a new `Block` and a ledger function to produce the new protocol parameters, we're fine.

Conversions between `SlotNo` and `UTCTime` will still require protocol params like  `SlotId`-`UTCTime` does. (but that's not the point here I believe)